tests/unit/task/test_base.py

```
@@ -0,0 +1,133 @@
+    
+    assert task.run_count == 1  # Only config2 should have run
+    assert len(results) == 2
+    assert results[0]["name"] == "test1"
+    assert results[1]["name"] == "test2"
+
+@pytest.mark.asyncio
+async def test_cache_directory_creation():
+    """Test that cache directory is created if it doesn't exist"""
+    cache_dir = Path("temp_test_cache")
+    try:
+        task = MockTask(cache_dir=cache_dir)
```

Ruff (0.8.2)

Local variable `task` is assigned to but never used

Remove assignment to unused variable `task`

(F841)